src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,30 +1,88 @@
-# TO-DO: Complete the selection_sort() function below 
-def selection_sort( arr ):
-    # loop through n-1 elements
-    for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
+import copy
+# TO-DO: Complete the selection_sort() function below
 
 
+def selection_sort(arr):
+    for i in range(0, len(arr) - 1):
+        start = arr[i]
+        # print(arr)
+        for j in range(i + 1, len(arr)):
+            if arr[i] > arr[j]:
+                arr[i], arr[j] = arr[j], arr[i]
 
-        # TO-DO: swap
+    # print(arr)
+    return arr
 
 
+# TO-DO:  implement the Bubble Sort function below
+def bubble_sort(arr):
+    def bubble_loop(arr):
+        progress = 0
+        for i in range(0, len(arr) - 1):
+            if arr[i] > arr[i+1]:
+                arr[i], arr[i+1] = arr[i+1], arr[i]
+            else:
+                progress += 1
+        return arr, progress
 
+    while True:
+        arr, progress = bubble_loop(arr)
+        # print(arr, progress)
+        if progress >= (len(arr) - 1):
+            break
 
```

Comment:
That is a really neat way to determine if the array is sorted!  That is, the use of the 'progress' variable and incrementing it each time you don't do a swap.  

Another similar approach would be to have some boolean value like `madeSwap` that you initially assume is false and you set true if you ever make a swap in that bubble loop.

Nice job!